Document MySQL row limit issue with Vitess (#48568)
metabase/metabase: docs/databases/connections/mysql.md

  command: ["--default-authentication-plugin=mysql_native_password"]
```

> ## Limitations with Vitess-based databases

Reviewer: ```suggestion
## When querying Vitess, you should limit your subqueries
```

> When using Metabase with MySQL databases which use Vitess as middleware, there can be issues with how row limits are handled. Typically, Metabase applies limits (e.g., 2000 or 10000 rows) to the final query results. However, due to a known bug in Vitess, these limits might be applied to subqueries instead, leading to unexpected results. To avoid this behavior, add a `LIMIT` clause inside every subquery manually to control the number of rows returned.

Reviewer: ```suggestion
When querying Vitess databases, you should add a `LIMIT` clause inside each subquery.

The reason: typically, Metabase applies limits (e.g., 2000 or 10000 rows) to the final query results. But due to a known bug in Vitess, Vitess might apply these limits to subqueries, which can lead to unexpected results. The workaround is to add limits to each of your subqueries.
```

Does this clarify it? I wanted to lead with what people should do, then explain why.
